### PR TITLE
amp-list: Support diffable with single-item

### PR DESCRIPTION
**Pull request by choumx**
Two fixes for `diffable`:
1. In #32650, we defer binding scan until render. But [this conditional](https://github.com/ampproject/amphtml/blob/master/extensions/amp-list/0.1/amp-list.js#L1106) is not the same as [this one](https://github.com/ampproject/amphtml/blob/df83808bc81014fc23fe9efb58f04bc91e1ff702/extensions/amp-list/0.1/amp-list.js#L1163). Make them the same.
2. With `single-item`, we shouldn't expect the diffable placeholder to have `[role=list]`.

`b/182456351`

### CODE REVIEW

**Comment by samouri** [comment]
Is this an alternative fix to: https://github.com/ampproject/amphtml/pull/33232?

**Comment by choumx** [comment]
Yup.

**Comment by samouri** [comment]
Just merged the revert, so lets combine the revert-revert with this PR. Let me know if you'd like me to take it over

**Comment by choumx** [comment]
Looks like @samouri is out for awhile, @caroqliu mind taking a look?

Not sure why CircleCI keeps getting cancelled. Is it still in use?

**Comment by kristoferbaxter** [comment]
If the PR hasn't pulled in latest main, Circle will cancel steps.

Recommend pulling in latest and then I'd be happy to review.

File: extensions/amp-list/0.1/amp-list.js [hunk 1]
```diff
@@ -1066,21 +1068,19 @@ export class AmpList extends AMP.BaseElement {
...
-            });
-          }
-        });
+        // This must be initial render, so do a non-blocking scan for bindings only.
+        // [diffable] is a special case that is handled later in render_(), see comment there.
+        if (!this.element.hasAttribute('diffable')) {
+          this.scanForBindings_(elements, []);
+        }
+
+        // Don't block render and return synchronously.
```
Comment (kristoferbaxter) [inline extensions/amp-list/0.1/amp-list.js:1083]: Thanks for adding comments here, much easier to follow the logic.

**Review by kristoferbaxter — APPROVED** [review]
Changes look great! Thanks.

**Comment by choumx** [comment]
> If the PR hasn't pulled in latest main, Circle will cancel steps.

This PR was originally based on `master` which caused failures. Rebasing off `main` fixed it. :) 

Thanks for the review!

**Pull request merged**